src/main/java/com/example/provider/json/SerializedLastLocation.java

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2022 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package com.example.provider.json;
+
+import com.google.protobuf.Int32Value;
+import com.google.auto.value.AutoValue;
+
+/** Representation of vehicle's last location. */
+@AutoValue
+abstract class SerializedLastLocation {
+
+  abstract SerializedLatLng latLng();
```

Comment:
Nit: I believe this should just be 'LatLng'

Reply:
Don't know why, but calling it 'LatLng()' will cause error in 'setLagLng'. It says 'error: Method does not correspond to a property ...'